### DISCUSSION TITLE
What do I have to do to not infringe UNO's licensing?

### DISCUSSION THREAD

**Discussion by DierkDroth** [unoplatform/uno#9554] [Q&A]
Here is my scenario:
- I'm developing a commercial, non-open source (=closed) application
- The application will incorporate a binary and unmodified copy of UNO Platform

I'm aware of [this](https://github.com/unoplatform/uno/blob/master/License.md), but am not clear what exactly I have to do to not infringe UNO's licensing. 

Please advise. Thanks in advance

**Comment by sasakrsmanovic** [comment]
Hello Dierk, 

The best thing to do is to contact a counsel to discuss the exact nature of your use of software which uses Apache 2.0 License. 

Google search shows some good resources on Apache 2.0 compliance; this one may offer some good insights https://fossa.com/blog/open-source-licenses-101-apache-license-2-0/. 

But at the end of the day, consulting someone in the legal field about your exact use case for using OSS Project with Apache 2.0 license would be the safest route.

Sasha

**Reply by DierkDroth** [reply]
@sasakrsmanovic thanks for your feedback and the link you provided.

However, I find it strange that UNO - as the holder of a license they chose - is unable to describe in a few simple words what exactly I had to do in my case - which I believe is a simple as possible one - in order to comply with said license. This clearly defeats the purpose of nventive's  OSS decision for UNO.

@francoistanguay FYI

**Comment by marwalsch** [comment]
@DierkDroth 

[This](https://tldrlegal.com/license/apache-license-2.0-(apache-2.0)) is arguably the most concise description of not only the capabilities Apache 2.0  grants you, but more importantly the responsibilities it demands (which are, since your distribution is unmodified, only marginally different from the less complex MIT license).

**Reply by DierkDroth** [reply]
Thanks for the link @marwalsch
